src/Vault.ts

```diff
@@ -8,6 +8,7 @@ import { resolveURL } from "./util";
 import { TotpVaultClient } from "./engines/totp";
 import { KVVaultClient } from "./engines/kv";
 import { KV2VaultClient } from "./engines";
+import * as fs from "fs/promises";
```

Comment:
This import fails in the workflow. Looks like `fs/promises` is only available in Node ≥14 and weirdly, some versions of Node 10 (https://github.com/nodejs/node/issues/35740#issuecomment-713778857).

But maybe the workflow could do with a bump to a halfway-recent Node version, anyway. 🤷‍♂️ 

LGTM, otherwise. :+1:

Reply:
Weird, I couldn't really figure out which api is recommended but switched to `import { promises as fs } from 'fs';` to keep compatibility with node 12+